source/upgrade/upgrading-mattermost-server.rst

```diff
@@ -81,87 +84,81 @@ Upgrading Mattermost Server
 
-9. You should first modify the last part to ``xargs echo rm -r`` to verify what will be executed. If you've added custom directories to the command in step b, then add those to the following command. For example:
-    
-.. code-block:: sh
+  The following command clears the contents of the ``mattermost`` folder, preserving only the specified directories and their contents. You should first modify the last part to ``xargs echo rm -r`` to verify what will be executed.
 
-  sudo find mattermost/ mattermost/client/ -mindepth 1 -maxdepth 1 \! \( -type d \( -path mattermost/client -o -path mattermost/client/plugins -o -path mattermost/config -o -path mattermost/logs -o -path mattermost/plugins -o -path mattermost/data \) -prune \) | sort | xargs echo rm -r
+  If you store TLSCert/TLSKey files or other information within your ``/opt/mattermost`` folder, you should add ``-o -path mattermost/yourFolderHere`` to the below command or you'll have to manually copy the TLSCert/TLSKey files from the backup into the new install.
 
-10. Clear the contents of this directory. If you've added custom directories to the command, be sure to add those to the following command. For example:
+  .. code-block:: sh
 
-.. code-block:: sh
+    sudo find mattermost/ mattermost/client/ -mindepth 1 -maxdepth 1 \! \( -type d \( -path mattermost/client -o -path mattermost/client/plugins -o -path mattermost/config -o -path mattermost/logs -o -path mattermost/plugins -o -path mattermost/data \) -prune \) | sort | sudo xargs rm -r  
+  
+  **Using Bleve Search**
 
-  sudo find mattermost/ mattermost/client/ -mindepth 1 -maxdepth 1 \! \( -type d \( -path mattermost/client -o -path mattermost/client/plugins -o -path mattermost/config -o -path mattermost/logs -o -path mattermost/plugins -o -path mattermost/data \) -prune \) | sort | sudo xargs rm -r
+  If using `Bleve Search <https://docs.mattermost.com/deploy/bleve-search.html>`__, and the directory exists *within* the ``mattermost`` directory, the index directory path won't be preserved with the commands below. 
+  
+  - You can either move the bleve index directory out from the ``mattermost`` directory before upgrading or, following an upgrade, you can copy the contents of the bleve index directory from the ``backup`` directory. 
+  - You can then store that directory or re-index as preferred. 
+  - The bleve indexes can be migrated without reindexing between Mattermost versions. See our `Configuration Settings <https://docs.mattermost.com/configure/configuration-settings.html#bleve-settings-experimental>`__ documentation for details on setting the bleve index directory.
       
-11. Change ownership of the new files before copying them. For example:
+8. Change ownership of the new files before copying them. For example:
 
-.. code-block:: sh
+  .. code-block:: sh
          
-  sudo chown -hR mattermost:mattermost /tmp/mattermost-upgrade/
+    sudo chown -hR mattermost:mattermost /tmp/mattermost-upgrade/
      
```

Comment:
In case this moves down: `sudo chown -R mattermost:mattermost {install-path}`. The `-h` option can be omitted and probably introduces more issues than it solves.